File: aoc-2023/src/test/java/com/capital7software/aoc/aoc2023/days/Day19Test.java
package com.capital7software.aoc.aoc2023.days;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

class Day19Test extends AdventOfCodeTestBase {
  private static final Logger LOGGER = LoggerFactory.getLogger(Day19Test.class);

  @BeforeEach
  void setUp() {
    var instance = new Day19();
    setupFromFile(instance.getDefaultInputFilename());
  }

  @Test
  void calculateAreaWithoutColors() {
    var instance = new Day19();
    var expected = 19_114;
    var actual = instance.sumOfRatingsOfAcceptedParts(lines);
    assertEquals(
        expected,
        actual,
        "The sum of the ratings of the accepted parts "
            + "is not what was expected: " + expected
    );
  }

  @Test
  void calculateAreaWithColors() {
    var instance = new Day19();
    var expected = 167_409_079_868_000L;
    var actual = instance.distinctCombinationsOfAcceptedRatings(lines);
    assertEquals(
        expected,
        actual,
        "The number of distinct combinations that will be accepted "
            + "is not what was expected: " + expected
    );
  }

  @Override
  protected Logger getLogger() {
    return LOGGER;
  }
}

File: aoc-2024/src/test/kotlin/com/capital7software/aoc/aoc2024aoc/days/Day19Test.kt
package com.capital7software.aoc.aoc2024aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day19Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day19Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day19()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testPossibleDesignCount() {
    val expected = 6L

    val actual = Day19().possibleDesignCount(lines)

    Assertions.assertEquals(
        expected,
        actual,
        "The possible number of designs $actual is not the expected value $expected."
    )
  }

  @Test
  fun testTotalArrangementCount() {
    val expected = 16L

    val actual = Day19().totalArrangementCount(lines)

    Assertions.assertEquals(
        expected,
        actual,
        "The number of arrangements $actual is not the expected value $expected."
    )
  }
}
